tests/snos.rs

```diff
@@ -30,6 +34,22 @@ fn print_a_hint(
     Ok(())
 }
 
+/// Mocks an os input but not a good way of doing things as it won't work in python.
+fn mock_os_input(
```

Comment:
let's move it to a `mock` module where we will add the scope mocking functionnality

Reply:
agreed. To handle in follow up.